tesseract-server/src/schema_config.rs

```diff
@@ -15,7 +16,38 @@ pub fn read_schema(schema_path: &String) -> Result<Schema, Error> {
     } else if schema_path.ends_with("json") {
         schema = Schema::from_json(&schema_str)?;
     } else {
-        panic!("Schema format not supported");
+        return Err(format_err!("Schema format not supported"))
+    }
+
+    // Check each cube for unique level and property names
+    for cube in schema.cubes.clone() {
+        let mut levels = HashSet::new();
+        let mut properties = HashSet::new();
+
+        for dimension in cube.dimensions.clone() {
+            for hierarchy in dimension.hierarchies.clone() {
+                for level in hierarchy.levels.clone() {
+                    if levels.contains(&level.name) {
```

Comment:
https://doc.rust-lang.org/1.0.0/std/collections/struct.HashSet.html#method.insert

I think that insert allows you to do this with one fn call, because it returns `false` if value is already in the set. (same goes for the `properties` check